### DISCUSSION TITLE
Yolov7 to Openvino

### DISCUSSION THREAD

**Discussion by georgethomas08** [openvinotoolkit/openvino#20884] [General]
Hi Team,

I'm a Novice player in Ai ML. I need a support to understand the step of converting yolov7 to openvino I'm following this document :https://docs.openvino.ai/2023.1/notebooks/226-yolov7-optimization-with-output.html. From this I have done upto (**Convert ONNX Model to OpenVINO Intermediate Representation (IR).)** . I need support to create inference pipeline, which is confusing for me with the steps mentioned in the document. My weights are trained with the custom class, so how do I need to procced? Is it mandatory that we need to do preprocessing and post processing to test the model in OpenVino

Regards
George Thomas

#support #20884

**Comment by yury-gorbachev** [comment]
Hi George,
 Model that you have converted was trained on certain input expectations, i.e. how input image is represented. This is part of training process. During inference, to convert to this representation you typically need to do some pre-processing outside of actual model inference. Same for output post-processing - you need to "decrypt" what model is giving you as a response, i.e. extract needed info from plain array of numbers.

 I see that you are reading tutorial, but if you have your model already, perhaps it will be easier for you to run this tutorial as a notebook within Jupyter environment and supply your model there. This is notebook: https://github.com/openvinotoolkit/openvino_notebooks/tree/main/notebooks/226-yolov7-optimization
 And you can check how to setup execution on your machine here: https://github.com/openvinotoolkit/openvino_notebooks/tree/main#-getting-started

 That way you can do step-by-step and see results right away.

Thanks

**Comment by georgethomas08** [comment]
Hi @yury-gorbachev ,
0)
Thank you for the information. I got the point but I'm confused, if you look to the notebook line( IN[10] ), it all about just to change the class name (DEFAULT_NAMES) with my class and run or do I need to change anything else? Secondly when it comes to In[11], what I need to change from the reference program?

**Reply by yury-gorbachev** [reply]
If you have same number of classes then likely you don't need to change anything except labels. But if number of classes is different then you need to update post-processing since output tensor directly depends on number of classes there.
It is not easiest case to advice on without knowing exact model details I have to say. Perhaps you could describe what you have done with the model? or how did you get it?

**Reply by georgethomas08** [reply]
Hi @yury-gorbachev ,

Sure, I have annotated 800 images 6 objects using labelImg, So numbering from the annotation text file will be like [ white:0, red:1, blue:2, damagedwhite:3, damagedred:4, damagedblue:5] I have followed the same order to define the classes yaml file in data and number 6 as in cfg yaml. in Yolov7

So in DEFAULT_NAMES = [ white, red, blue, damagedwhite, damagedred, damagedblue ], I got up-to this point, apart from this do I need to add or change any parameters

**Reply by yury-gorbachev** [reply]
So it looks like you have 6 classes and you need to fix post-processing code that assumes that there are 80 classes in output. You can read comments in notebook and see what format is for this tensor.
It is one of the hardest parts in general when adopting model since you need to account for format that was designed by model author. I recommend to output tensor via python and check shape and contents to see that it matches your expectations and after that fix post-processing code.
